tree/dataframe/inc/ROOT/RDF/RDefine.hxx

```diff
@@ -157,12 +157,16 @@ public:
    /// Create clones of this Define that work with values in varied "universes".
    void MakeVariations(const std::vector<std::string> &variations) final
    {
-      // the varied defines get a copy of the callable object.
-      // TODO document this
       for (const auto &variation : variations) {
+         if (std::find(fVariationDeps.begin(), fVariationDeps.end(), variation) == fVariationDeps.end()) {
```

Comment:
I suppose fVariationDeps is including any upstream dependency, i.e. not just immediate ones?

Reply:
yep!